### DISCUSSION TITLE
Using Servo as a browser and tracking the project's status and development

### DISCUSSION THREAD

**Discussion by BlueNebulaDev** [servo/servo#28608] [General]
I have a couple of questions emerging from my wish to use Servo as a web browser, instead of Chromium or Firefox.

1. Is there any project developing an UI for Servo? Servo itself comes with a binary, but it's too minimal: no navigation or search bar, no tabs, no bookmarks, no context menu...  
  Shouldn't it be somewhat simple to develop and maintain a similar interface? Is anybody attempting it?

2. The aforementioned Servo binary has some issues... It takes a long time (seconds) to render most pages (eg after minimizing and restoring the window), most are not rendered correctly, the scroll behavior is buggy, and so on.
    1. What's causing these issues? Which components are buggy or incomplete? Both Stylo and WebRender are used by Firefox and they don't have these issues there...
    2. Is there any page tracking the status of Servo and grouping the various issues? This repository has 3,320 open issues and it's impossible for me to understand what works, what doesn't, which features are not yet implemented, which issues block what, and so on.
    To make some examples, how can I find out which issues impair the correct rendering of https://en.wikipedia.org/wiki/Main_Page? Or how can I find out which properties of CSS 2.1 are fully supported?

**Comment by emmyoh** [comment]
Servo was not originally intended to be a browser (I believe? Could be wrong); the project was created to rewrite some Firefox components in Rust.
I believe the current goal, since its transfer from Mozilla to the Linux Foundation, is to allow for the usage of Servo as an embedded web view. I'm not sure what work has actually been done on this, however—there are no ports to GTK, Qt, or other UI frameworks AFAIK.
So, why doesn't Servo come with a UI? It's not meant to have one. Why hasn't a browser been built using Servo as its sole browser engine? It hasn't been made to work with any pre-existing UI toolkits (praying for a Servo GTK port myself).

**Reply by niutech** [reply]
There is a [libsimpleservo](https://github.com/servo/servo/tree/master/ports/libsimpleservo) port, which you could embed in your app.

**Comment by emmyoh** [comment]
For an understanding of what Servo supports—I'm not aware of any 'status' page breaking down what is and is not implemented. However, using Servo, you could navigate to [html5test.com](https://html5test.com/) to see yourself what HTML5 features Servo implements (that the website tests for).
